File: dsa-interview-crash-course/2270_number_of_ways_to_split_array.py
from typing import List


class Solution:
    def waysToSplitArray(self, nums: List[int]) -> int:
        prev = 0
        prefix = []
        for i in range(len(nums)):
            prefix.append(prev + nums[i])
            prev = prefix[i]
        
        num_ways = 0
        for i in range(len(nums)-1):
            if (prefix[i] >= (prefix[-1] - prefix[i])):
                num_ways += 1
        
        return num_ways

File: dsa-interview-crash-course/2270_number_of_ways_to_split_array.cpp
#include <vector>

class Solution {
   public:
    int waysToSplitArray(std::vector<int>& nums) {
        std::vector<long> prefix(nums.size() + 1, 0);

        for (int i = 0; i < nums.size(); ++i) {
            prefix[i + 1] = static_cast<long>(nums[i]) + prefix[i];
        }

        int num_ways = 0;
        for (int i = 1; i < prefix.size() - 1; ++i) {
            if (prefix[i] >= prefix.back() - prefix[i]) {
                ++num_ways;
            }
        }

        return num_ways;
    }
};